app/controllers/reports/regions_controller.rb

```diff
@@ -206,4 +210,36 @@ def percentage(numerator, denominator)
     return 0 if denominator == 0 || numerator == 0
     ((numerator.to_f / denominator) * 100).round(2)
   end
+
+  def set_data_for_who_report
+    @months = @period.downto(5).reverse
+    repo = Reports::Repository.new(@region, periods: @period)
```

Comment:
You can use the same repo for all the data here by doing:

```
regions = @region.facility_regions << @region
Repository.new(regions, periods: @period)
```

that will also reuse the same memorized cached, so better perf.

Reply:
Ah, thanks for the tip. I tried something similar but failed because the child facilities were an active record relation.